File: VB/SimpleDataSet.vb
Imports System
Imports System.ComponentModel
Imports System.Data
Imports System.Windows
Imports System.Windows.Data

Namespace WPFPrintListView

    Public Class SimpleDataSet
        Inherits DataSet

        Public Sub New()
            MyBase.New()
            Dim table As DataTable = New DataTable("table")
            DataSetName = "ManualDataSet"
            table.Columns.Add("ID", GetType(Integer))
            table.Columns.Add("MyDateTime", GetType(Date))
            table.Columns.Add("MyRow", GetType(String))
            table.Columns.Add("MyData", GetType(Double))
            table.Constraints.Add("IDPK", table.Columns("ID"), True)
            Tables.AddRange(New DataTable() {table})
        End Sub

        Public Shared Function CreateData() As SimpleDataSet
            Dim ds As SimpleDataSet = New SimpleDataSet()
            Dim table As DataTable = ds.Tables("table")
            Dim rnd As Random = New Random(Date.Now.Millisecond)
            For i As Integer = 0 To 100 - 1
                table.Rows.Add(New Object() {i, Date.Today.AddDays(1), If(i Mod 2 = 0, "A", "B"), Math.Round(rnd.NextDouble() * 100, 2)})
            Next

            Return ds
        End Function

        <DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)>
        Public Overloads ReadOnly Property Tables As DataTableCollection
            Get
                Return MyBase.Tables
            End Get
        End Property

        <DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)>
        Public Overloads ReadOnly Property Relations As DataRelationCollection
            Get
                Return MyBase.Relations
            End Get
        End Property
    End Class

    <ValueConversion(GetType(Date), GetType(String))>
    Public Class DateConverter
        Implements IValueConverter

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As Globalization.CultureInfo) As Object Implements IValueConverter.Convert
            Dim [date] As Date = CDate(value)
            Return [date].ToShortDateString()
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
            Dim strValue As String = TryCast(value, String)
            Dim resultDateTime As Date
            If Date.TryParse(strValue, resultDateTime) Then
                Return resultDateTime
            End If

            Return DependencyProperty.UnsetValue
        End Function
    End Class
End Namespace

File: CS/SimpleDataSet.cs
using System;
using System.ComponentModel;
using System.Data;
using System.Windows;
using System.Windows.Data;

namespace WPFPrintListView
{
    public class SimpleDataSet : DataSet
    {
        public SimpleDataSet()
            : base()
        {
            DataTable table = new DataTable("table");

            DataSetName = "ManualDataSet";

            table.Columns.Add("ID", typeof(Int32));
            table.Columns.Add("MyDateTime", typeof(DateTime));
            table.Columns.Add("MyRow", typeof(string));
            table.Columns.Add("MyData", typeof(double));
            table.Constraints.Add("IDPK", table.Columns["ID"], true);

            Tables.AddRange(new DataTable[] { table });
        }

        public static SimpleDataSet CreateData()
        {
            SimpleDataSet ds = new SimpleDataSet();
            DataTable table = ds.Tables["table"];

            Random rnd = new Random(DateTime.Now.Millisecond);

            for (int i = 0; i < 100; i++)
                table.Rows.Add(new object[] { i, DateTime.Today.AddDays(1), (i % 2 == 0 ? "A" : "B"), Math.Round(rnd.NextDouble() * 100, 2) });

            return ds;
        }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
        public new DataTableCollection Tables
        {
            get { return base.Tables; }
        }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
        public new DataRelationCollection Relations
        {
            get { return base.Relations; }
        }
    }

    [ValueConversion(typeof(DateTime), typeof(String))]
    public class DateConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            DateTime date = (DateTime)value;
            return date.ToShortDateString();
        }

        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            string strValue = value as string;
            DateTime resultDateTime;
            if (DateTime.TryParse(strValue, out resultDateTime))
            {
                return resultDateTime;
            }
            return DependencyProperty.UnsetValue;
        }
    }

}
